src/internal/connector/onedrive/collection.go

```diff
@@ -122,6 +135,26 @@ func NewCollection(
 	return c
 }
 
+// Figures out the state of a collection using prev and current path
+// This is similar to the one in exchange, but instead of checking the
+// equivalence of folders, it checks it for the actual item.
+// TODO(meain): Why is exchange checking folders?
+func stateOf(prev, curr path.Path) data.CollectionState {
```

Comment:
This can be done in a follow up PR, but we should centralize this function since it's probably going to be used a lot. I guess maybe we can put it in the `data` package since it's independent from the service making the collections?

The Exchange version of this function also has tests associated with it which will need to be moved

Reply:
Fixed it, https://github.com/alcionai/corso/pull/2407#discussion_r1097737786 was the culprit.